src/Corax/Indexing/IndexWriter.cs

```diff
@@ -1285,6 +1330,136 @@ public void InsertTextualField()
                 return totalLengthOfTerm;
             }
             
+            void ProcessTermsVector()
+            {
+                if (_indexedField.FieldIndexingMode is not FieldIndexingMode.Search)
+                    return;
+
+                const StoredFieldType storedFieldType = (StoredFieldType.List | StoredFieldType.Term);
+                _writer.InitializeFieldRootPageForTermsVector(_indexedField); 
+                var termsPerEntrySpan = _writer._termsPerEntryId.ToSpan();
+                
+                IDisposable memoryHandler = null;
+                var processingBufferPosition = 0;
+                var virtualMapping = _virtualTermIdToTermContainerId.ToSpan();
+                
+                Span<long> termsBuffer = stackalloc long[32];
+                Span<int> indexesBuffer = stackalloc int[32];
+                Span<byte> processingBuffer = stackalloc byte[32 * ZigZagEncoding.MaxEncodedSize];
+                
+                for (var documentIndex = 0; documentIndex < _indexedField.EntryToTerms.Count; ++documentIndex)
+                {
+                    ref var fieldTerms = ref _indexedField.EntryToTerms[documentIndex];
+                    ref var entryTerms = ref termsPerEntrySpan[documentIndex];
+                    
+                    //When document has no terms we proceed
+                    if (fieldTerms.Count == 0)
+                        continue;
+
+                    if (fieldTerms.Count > termsBuffer.Length)
+                        UnlikelyGrowBuffer(_writer._entriesAllocator, fieldTerms.Count, ref termsBuffer, ref indexesBuffer, ref processingBuffer);
+
+                    var terms = termsBuffer.Slice(0, fieldTerms.Count);
+                    var indexes = indexesBuffer.Slice(0, fieldTerms.Count);
+                    
+                    for (var termIndex = 0; termIndex < fieldTerms.Count; ++termIndex)
+                    {
+                        ref var virtualTermId = ref fieldTerms[termIndex];
+                        Debug.Assert(virtualMapping.Length > virtualTermId, "_indexedField.NativeVirtualTermIdToTermContainerId.Count > term");
+
+                        terms[termIndex] = virtualMapping[virtualTermId];
+                        indexes[termIndex] = termIndex << 1; // Gives bit for duplicate marker.
```

Comment:
Do we have to worry about batches where we index > 1B terms?
We are using ints, so limited to 2.1 B terms in a batch, and we lose a bit, so 1B max now.

I know it is ridiculous, but let's add an explicit check that would prevent it.